lib/alephant/broker/response/batch.rb

```diff
@@ -28,18 +31,40 @@ def json
           logger.info "Broker: Batch load started (#{batch_id})"
           components.map do |component|
             {
-              'component'    => component.id,
-              'options'      => component.options,
-              'status'       => component.status,
-              'content_type' => component.content_type,
-              'body'         => component.content
+              "component"    => component.id,
+              "options"      => component.options,
+              "status"       => component.status,
+              "content_type" => component.content_type,
+              "body"         => component.content
             }
-          end.tap { 
-            logger.info "Broker: Batch load done (#{batch_id})" 
+          end.tap {
+            logger.info "Broker: Batch load done (#{batch_id})"
             logger.metric "BrokerBatchLoadCount"
           }
         end
 
+        def batch_response_headers
+          {
+            "ETag"          => batch_response_etag,
+            "Last-Modified" => batch_response_last_modified
+          }
+        end
+
+        def batch_response_etag
+          etags = components.map do |component|
+            component.headers["ETag"] if component.headers["ETag"]
+          end
```

Comment:
I would add a `.compact` there as you're possibly going to get some `nil` values in your array (due to the use of the `if`) - infact, you could probably just remove the `if` and just use `.compact` if you liked.